core/src/test/java/org/elasticsearch/deps/joda/SimpleJodaTests.java

```diff
@@ -301,16 +285,26 @@ public void testThatNegativeEpochsCanBeParsed() {
             assertThat(dateTime.getSecondOfMinute(), is(20));
         }
 
+        // test floats get truncated
+        String epochFloatValue = String.format(Locale.US, "%d.%d", dateTime.getMillis() / (parseMilliSeconds ? 1L : 1000L), randomNonNegativeLong());
```

Comment:
It **is** negative since `dateTime.getMillis()` is negative, the non negative long is for the fractional part.